src/core/service.c

```diff
@@ -1358,10 +1372,77 @@ static int service_collect_fds(
                 rfd_names[n_fds] = NULL;
         }
 
+        if (s->open_files) {
+                size_t n_fds = rn_socket_fds + s->n_fd_store;
+
+                LIST_FOREACH(open_files, f, s->open_files) {
+                        struct stat st;
+                        int fd = -1;
+
+                        if (stat(f->path, &st) < 0) {
+                                log_error_errno(errno, "Failed to stat %s: %m", f->path);
+                                continue;
+                        }
```

Comment:
We should probably return the error here if the file doesn't exist by default.

Maybe in the future we can add an option that allows the file to not exist, but by default let's require the file to exist.